src/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandlePropertyName.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Collections;
+using System.Diagnostics;
+using System.Linq;
+
+namespace System.Text.Json.Serialization
+{
+    public static partial class JsonSerializer
+    {
+        private static void HandlePropertyName(
+            JsonSerializerOptions options,
+            ref Utf8JsonReader reader,
+            ref ReadStack state)
+        {
+            if (state.Current.Drain)
+                return;
+
+                Debug.Assert(state.Current.ReturnValue != default);
+                Debug.Assert(state.Current.JsonClassInfo != default);
+
+            if (state.Current.IsProcessingDictionary)
+            {
+                if (ReferenceEquals(state.Current.JsonClassInfo.DataExtensionProperty, state.Current.JsonPropertyInfo))
+                {
+                    ReadOnlySpan<byte> propertyName = reader.HasValueSequence ? reader.ValueSequence.ToArray() : reader.ValueSpan;
+                    if (reader._stringHasEscaping)
```

Comment:
Can you add a TODO comment pointing to https://github.com/dotnet/corefx/issues/35386?